src/App.tsx

```diff
@@ -1,223 +1,196 @@
-                className="todo__status"
-              />
-            </label>
+  const visibleTodos = useMemo(() => {
+    return filterBySelect(todos, filtredBy);
+  }, [todos, filtredBy]);
 
-            <span data-cy="TodoTitle" className="todo__title">CSS</span>
+  const amountOfActiveTodos = useMemo(() => {
+    return filterBySelect(todos, FilterOptions.ACTIVE).length;
+  }, [todos]);
 
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <form>
-              <input
-                data-cy="TodoTitleField"
-                type="text"
-                className="todo__title-field"
-                placeholder="Empty todo will be deleted"
-                defaultValue="JS"
-              />
-            </form>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
+  const isClearNeeded = useMemo(() => {
+    return amountOfActiveTodos !== visibleTodos.length;
+  }, [amountOfActiveTodos, visibleTodos]);
 
-            <span data-cy="TodoTitle" className="todo__title">React</span>
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
+  const compTodos = todos
+    .filter(todo => todo.completed);
+
+  const clearCompletedTodos = async () => {
+    await Promise.all(compTodos
+      .map((todo) => deleteCurrentTodo(todo.id)));
```

Comment:
you don't need a new array here 🙃 
```suggestion
      .forEach((todo) => await deleteCurrentTodo(todo.id)));
```